extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java

```diff
@@ -701,33 +703,72 @@ private static void contributeBuiltinConstraints(Set<String> builtinConstraints,
         }
     }
 
-    private static void contributeClass(Set<DotName> classNamesCollector, IndexView indexView, DotName className) {
-        classNamesCollector.add(className);
+    private static void contributeClass(Set<DotName> classNamesCollector, IndexView indexView, ClassInfo classInfo) {
+        if (isInGraalVMFeature(indexView, classInfo)) {
+            return;
+        }
 
-        if (DotNames.OBJECT.equals(className)) {
+        classNamesCollector.add(classInfo.name());
+
+        if (DotNames.OBJECT.equals(classInfo.name())) {
             return;
         }
 
-        for (ClassInfo subclass : indexView.getAllKnownSubclasses(className)) {
+        for (ClassInfo subclass : indexView.getAllKnownSubclasses(classInfo.name())) {
             if (Modifier.isAbstract(subclass.flags())) {
                 // we can avoid adding the abstract classes here: either they are parent classes
                 // and they will be dealt with by Hibernate Validator or they are child classes
                 // without any proper implementation and we can ignore them.
                 continue;
             }
+            if (isInGraalVMFeature(indexView, subclass)) {
+                return;
+            }
             classNamesCollector.add(subclass.name());
         }
-        for (ClassInfo implementor : indexView.getAllKnownImplementors(className)) {
+        for (ClassInfo implementor : indexView.getAllKnownImplementors(classInfo.name())) {
             if (Modifier.isAbstract(implementor.flags())) {
                 // we can avoid adding the abstract classes here: either they are parent classes
                 // and they will be dealt with by Hibernate Validator or they are child classes
                 // without any proper implementation and we can ignore them.
                 continue;
             }
+            if (isInGraalVMFeature(indexView, implementor)) {
```

Comment:
I was thinking whether we could say something like:
```
if (hasAnyConstraints(indexView, implementor)) {
```
If there are no constraints, there's nothing to cascade into, right? But I don't remember if we need to keep the classes without constraints.. will try to take a look later

Reply:
I think:
- if the class is a leaf class
- it doesn't have any constraints
we can skip it.

So I think we could go with your approach... but... I would still try to exclude the `Feature` classes as you never know if someone adds a `@NotNull` by mistake.

Reply:
yeah +1 just to be safe

Reply:
yes 👍🏻 🙂

Reply:
ok checked aand that won't fly 🙈:
```java
Validator validator = validator( Set.of( A.class ) );

assertThat( validator.validate( new A() ) ).containsOnlyViolations(
		violationOf( NotNull.class ).withProperty( "name" )
);

// fails as we get the `UninitializedBeanMetaData` and hence no constraints ... 
assertThat( validator.validate( new B() ) ).containsOnlyViolations(
		violationOf( NotNull.class ).withProperty( "name" )
);
```
If we skip the class, even if it's a leaf with no constraints (or if it's anywhere in the hierarchy for that matter), we will end up having the `UninitializedBeanMetaData` for it once someone wants to validate an instance of it and it simply has 0 constraints (the ones from the super classes are there).

Reply:
I think we should have started with: WTH is this class in the index? I wouldn't expect it to be.